lib.rs

```diff
@@ -148,15 +148,49 @@ fn error(e: &str) -> TokenStream {
     TokenStream::from_str(&format!("::core::compile_error!{{\"{}\"}}", e.escape_default())).unwrap()
 }
 
+fn parse_feature_label(input: syn::parse::ParseStream) -> syn::parse::Result<String> {
+    // parse the key, which must be an identifier
+    let ident: syn::Ident = input.parse()?;
+
+    // ensure that the identifier is `feature_label`
+    if ident != "feature_label" {
+        // Trigger an error not on the current position of the stream,
+        // but on the position of the unexpected identifier.
+        return Err(syn::Error::new(ident.span(), "expected `feature_name`"));
+    }
+
+    // parse a single equal sign `=`
+    input.parse::<syn::Token![=]>()?;
+
+    // parse the value, which must be a string literal
+    let literal: syn::LitStr = input.parse()?;
+    // get the string itself
+    let value = literal.value();
+
+    // ensure that the string contains the substring `"{feature}"`
+    if value.find("{feature}").is_none() {
+        return Err(syn::Error::new(
+            literal.span(),
+            "expected a string literal which contains the substring \"{feature}\"",
+        ));
+    }
+    Ok(value)
+}
+
 /// Produce a literal string containing documentation extracted from Cargo.toml
 ///
 /// See the [crate] documentation for details
 #[proc_macro]
-pub fn document_features(_: TokenStream) -> TokenStream {
-    document_features_impl().unwrap_or_else(std::convert::identity)
+pub fn document_features(tokens: TokenStream) -> TokenStream {
+    if tokens.is_empty() {
+        document_features_impl(None).unwrap_or_else(std::convert::identity)
+    } else {
+        let feature_label = syn::parse_macro_input!(tokens with parse_feature_label);
+        document_features_impl(Some(&feature_label)).unwrap_or_else(std::convert::identity)
+    }
 }
 
-fn document_features_impl() -> Result<TokenStream, TokenStream> {
+fn document_features_impl(feature_label: Option<&str>) -> Result<TokenStream, TokenStream> {
```

Comment:
Maybe have something like

```rust
struct Args {
  feature_label: Option<&str>,
}
```

In case we end up adding more option in the future.

Reply:
Can I leave it as is for the moment? I think this refactoring can go in the next PR that adds some other arguments.

Reply:
ok